### PR TITLE
Add the L2 distance layer.

### PR DESCRIPTION
**Pull request by lcy-seso**
fixes https://github.com/PaddlePaddle/Paddle/issues/5739

### CODE REVIEW

File: python/paddle/trainer_config_helpers/layers.py [hunk 1]
```diff
@@ -2333,6 +2335,51 @@ def cos_sim(a, b, scale=1, size=1, name=None, layer_attr=None):
 
 
 @wrap_name_default()
+@layer_support()
+def l2_distance_layer(x, y, name=None, layer_attr=None):
+    """
+    This layer calculate and return the Euclidean distance between two input
```
Comment (ranqiu92) [inline python/paddle/trainer_config_helpers/layers.py]: ...calculates...returns...

File: python/paddle/trainer_config_helpers/layers.py [hunk 2]
```diff
@@ -2333,6 +2335,51 @@ def cos_sim(a, b, scale=1, size=1, name=None, layer_attr=None):
...
+    :type layer_attr: ExtraLayerAttribute
+    :return: The returned LayerOutput object.
+    :rtype: LayerOutput
+    """
+
+    assert isinstance(x, LayerOutput) and isinstance(x, LayerOutput)
+    Layer(
+        name=name,
+        type=LayerType.L2_DISTANCE,
+        inputs=[x.name, x.name],
```
Comment (ranqiu92) [inline python/paddle/trainer_config_helpers/layers.py]: y.name

File: python/paddle/trainer/config_parser.py [hunk 3]
```diff
@@ -3330,6 +3330,18 @@ def __init__(self, name, inputs, **xargs):
...
 
+@config_layer('cos')
+class CosSimLayer(LayerBase):
+    def __init__(self, name, inputs, cos_scale=1, device=None):
+        super(CosSimLayer, self).__init__(
+            name, 'cos', 1, inputs=inputs, device=device)
+        config_assert(len(self.inputs) == 2, 'CosSimLayer must have 2 inputs')
+        config_assert(
+            self.get_input_layer(0).size == self.get_input_layer(1).size,
+            'inputs of CosSimLayer must have same dim')
```
Comment (ranqiu92) [inline python/paddle/trainer/config_parser.py]: 首字母大写，dim单词不要简写

Comment (lcy-seso) [reply python/paddle/trainer/config_parser.py]: Done.

File: python/paddle/trainer_config_helpers/layers.py (hunk 1 above)
Comment (lcy-seso) [reply python/paddle/trainer_config_helpers/layers.py]: Done.

File: python/paddle/trainer_config_helpers/layers.py (hunk 2 above)
Comment (lcy-seso) [reply python/paddle/trainer_config_helpers/layers.py]: Done, thank you.

**Pull request merged**